lib/mssql.js

```diff
@@ -163,8 +134,22 @@ export async function check(req, res, pool) {
   return {ok: true};
 }
 
+export function addToPools(config, pools, {signal}) {
+  const pool = new mssql.ConnectionPool(config);
+  const close = pool.close.bind(pool);
+
+  pool.close = (...args) => {
+    pools.delete(JSON.stringify(config));
```

Comment:
They key should be created only once and then kept in the closure, in case the config object gets mutated.